src/main/java/seedu/teachbook/ui/MainWindow.java

```diff
@@ -138,6 +138,9 @@ void fillInnerParts() {
     void updateStudentListPanel() {
         personListPanel = new PersonListPanel(logic.getFilteredPersonList());
         personListPanelPlaceholder.getChildren().add(personListPanel.getRoot());
+
+        GeneralIndex newSelectedClassIndex = logic.getCurrentlySelectedClassIndex();
+        classListPanel.reload(newSelectedClassIndex.getZeroBased());
```

Comment:
Hmm actually now student and class list panel can and should be updated separately. Use the method below? (So these two lines can be deleted.)